GVFS/GVFS.Common/Git/ICredentialManager.cs

```diff
@@ -0,0 +1,13 @@
+﻿using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public interface ICredentialManager

```

Comment:
To me the simpler the interface the better.  I would imagine in a perfect world it would be something more like a database CRUD:
```
public class CredentialData
{
    public string Url { get; set; }
    public string Username { get; set; }
    public string Password { get; set; }
}

public interface ICredentialManager
{
    CredentialData GetCredential();
    void StoreCredential(CredentialData data);
    void DeleteCredential(CredentialData data);
}
```

And you could use exceptions or return `bool` instead of `void` but this is just me off in a perfect world.

Reply:
After writing that it makes sense to name the interface `ICredentialStore`

Reply:
I really like this API as an end goal